diff --git a/admin_toolbar_search/admin_toolbar_search.module b/admin_toolbar_search/admin_toolbar_search.module
index acad2f2..fa7ce8c 100755
--- a/admin_toolbar_search/admin_toolbar_search.module
+++ b/admin_toolbar_search/admin_toolbar_search.module
@@ -28,13 +28,14 @@ function admin_toolbar_search_help($route_name, RouteMatchInterface $route_match
  * Implements hook_toolbar_alter().
  */
 function admin_toolbar_search_toolbar_alter(&$items) {
-  $access = \Drupal::currentUser()->hasPermission('use admin toolbar search');
+  if (!\Drupal::currentUser()->hasPermission('use admin toolbar search')) {
+    return;
+  }
   $admin_toolbar_tools_enabled = \Drupal::service('module_handler')
     ->moduleExists('admin_toolbar_tools');
 
   $items['administration_search'] = [
     "#type" => "toolbar_item",
-    '#access' => $access,
     'tab' => [
       '#type' => 'link',
       '#title' => new TranslatableMarkup('Search'),
