diff --git a/src/Controller/AutobanController.php b/src/Controller/AutobanController.php
index 98b7dbc..5f15125 100644
--- a/src/Controller/AutobanController.php
+++ b/src/Controller/AutobanController.php
@@ -187,21 +187,20 @@ class AutobanController extends ControllerBase {
     $container = \Drupal::getContainer();
     $kernel = $container->get('kernel');
 
-    // Get all services list.
-    $services = $kernel->getCachedContainerDefinition()['services'];
-    foreach ($services as $service_id => $value) {
-      $service_def = unserialize($value);
-      if (!empty($service_def['properties']) && !empty($service_def['properties']['_serviceId'])) {
-        $service_id = $service_def['properties']['_serviceId'];
-        $aservices = explode('.', $service_id);
+    $services = array_filter(
+      // Get all services list.
+      array_keys($kernel->getCachedContainerDefinition()['services']),
+      function ($service_id) {
         // Filter for services with ban_provider.
-        if (!empty($aservices[1]) && $aservices[1] == 'ban_provider') {
-          $service = $container->get($service_id);
-          $id = $service->getId();
-          $name = $service->getName();
-          $banProvidersList[$id] = ['name' => $name, 'service' => $service];
-        }
+        return preg_match('/^[^.]+\.ban_provider$/', $service_id) === 1;
       }
+    );
+
+    foreach ($services as $service_id) {
+      $banProvidersList[($service = $container->get($service_id))->getId()] = [
+        'name' => $service->getName(),
+        'service' => $service,
+      ];
     }
 
     if (!empty($provider_id)) {
