diff --git a/js/recaptcha.invisible.js b/js/recaptcha.invisible.js
index 589ab00..3c5ba1c 100644
--- a/js/recaptcha.invisible.js
+++ b/js/recaptcha.invisible.js
@@ -94,7 +94,7 @@ function recaptchaOnInvisibleSubmit(token) {
       function preventFormSubmit(elem, event) {
         if (grecaptcha.getResponse().length === 0) {
           // We need validate form, to avoid prevention of html5 validation.
-          var form = $(elem).closest('form')[0];
+          var form = $(elem).addClass('focus').closest('form')[0];
           if (form && typeof form.checkValidity === 'function' && !$(form).attr('validate')) {
             if (form.checkValidity()) {
               event.preventDefault();
